include/cis1_proto_utils/event_dispatcher.h

```diff
@@ -150,9 +165,21 @@ class event_dispatcher
         return error::invalid_json;
     }
 
+    static void on_error(
+            std::optional<std::function<error_handler_t>> error_handler,
+            std::string_view desc)
+    {
+        if(error_handler)
+        {
+            error_handler.value()(desc);
+        }
+    }
+
     std::map<std::string, std::function<event_handler_t>> event_handlers_;
+    std::optional<std::function<error_handler_t>> error_handler_ = std::nullopt;
```

Comment:
Maybe better use just std::function instead of std::optional<std::function<...>>? You still can check is handler set with conversion to bool operator `if(error_handler_) ...` and can reset with copy assignment operator `error_handler_ = {};`

Reply:
Done